dnf/conf/substitutions.py

```diff
@@ -60,7 +63,11 @@ def update_from_etc(self, installroot, varsdir=("/etc/yum/vars/", "/etc/dnf/vars
                             val = fp.readline()
                         if val and val[-1] == '\n':
                             val = val[:-1]
-                    except (OSError, IOError):
+                    except (OSError, IOError, UnicodeDecodeError) as e:
+                        fsvar_error = True
+                        logger.debug(_("Error when processing variable file '%s': %s"), filepath, e)
```

Comment:
Perfect, the message contains all required information. The message is fine but what about a minor change:
"Error when setting a variable from file '%s': %s"
"Error when parsing a variable from file '%s': %s"

Reply:
Using `'{0}': {1}` allows translators to use arguments in a different order.